crates/rpc/src/server/mod.rs

```diff
@@ -11,44 +10,149 @@ use crate::COMPONENT;
 
 mod api;
 
-/// Represents an initialized rpc component where the RPC connection is open, but not yet actively
-/// responding to requests.
-///
-/// Separating the connection binding from the server spawning allows the caller to connect other
-/// components to the store without resorting to sleeps or other mechanisms to spawn dependent
-/// components.
```

Comment:
Would be good to keep a general overview of how it is initialized, what are the expectations, etc.